src/Symfony/Component/Form/Extension/Core/Type/TextType.php

```diff
@@ -27,6 +27,9 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         // See https://github.com/symfony/symfony/issues/5906#issuecomment-203189375
         if ('' === $options['empty_data']) {
             $builder->addViewTransformer($this);
+        } elseif (FormType::$emptyData === $options['empty_data']) {
+            // "empty_data" option is the default value from FormType::configureOptions()
+            trigger_deprecation('symfony/form', '5.4', 'The default value of "empty_data" option in "%s" will be changed to empty string. Declare "NULL" as value for "empty_data" if you still want use "NULL" as data.', __CLASS__);
```

Comment:
To ensure that this is the default closure and not the user-specified one, we must compare the provided closure with the default closure. To do this, the default closure must be created once and be accessible from the outside.